### PR TITLE
[Refactor] Closest Interactable Calculation

### LINKED ISSUE #214: [Refactor] Closest interactable called on every update (+ titipan)
Fix with player -> NearbyItems
NearbyItems class is similar to inventory.
NearbyItems nyimpen item mana aja yang lagi di zona player.
Kalo player udah keluar dari zona item, keluarin item dari list NearbyItems

Still call on every update but no longer cast on every frame

Rename _movementSpeed in PlayerBase

Refactor, put _detectorZone in abstract. Fix Start() to Awake()

Raname funcs in ObjectDetector

### LINKED ISSUE #215: [Refactor] Adjust debug log tag messages with drawio systems

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- Refactor closest interactable calculation
- Adjust trigger area of interactables
- Adjust debug log message format
- Reorder regions

### Tasks List
- [close #214 ]
- [close #215 ]

### Attachments
IMG_0358

### CODE REVIEW

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Overall its already good refactors, some small changes are required though.

Also added note:
ALL functions should stay in a region (even if thet region currently only have one function)

File: Assets/Astralization/Scripts/UI/ItemLogo.cs [hunk 1]
```diff
@@ -6,16 +6,22 @@
...
+    #region Variables
     private Image _img;
+    #endregion
 
-    private void Awake()
-    {
-        _img = transform.Find("Logo").GetComponent<Image>();
-    }
-
+    #region SetGet
```
[the remark this answers is not in the document]
Comment (rafiftaris) [reply Assets/Astralization/Scripts/UI/ItemLogo.cs:13]: i think UpdateLogo works like setting visibility state and its sprite, so it is a setter

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

**Review by raihansyahdean — APPROVED** [review]
Well done!

**Pull request merged**